utils/train.py

```diff
@@ -13,18 +12,35 @@
 from .validation import validate
 
 
+def cosine_decay(init_val, final_val, step, decay_steps):
+    alpha = final_val / init_val
```

Comment:
Shouldn't this be `init_val / final_val`?

Reply:
The learning rate decays. You might write a demo for testing.
```python
init_val = 1e-4
final_val = 1e-5
```

Reply:
According to the following source it's **"Minimum learning rate value as a fraction of learning_rate."**
https://docs.w3cub.com/tensorflow~python/tf/train/cosine_decay/

So yes, they should be switched around.